scrapy/utils/reqser.py

```diff
@@ -90,10 +92,12 @@ def _find_method(obj, func):
             pass
         else:
             if func_self is obj:
-                name = func.__func__.__name__
-                if _is_private_method(name):
-                    return _mangle_private_name(obj, func, name)
-                return name
+                members = inspect.getmembers(obj, predicate=inspect.ismethod)
+                for name, obj_func in members:
+                    if obj_func.__func__ is func.__func__:
+                        if _is_private_method(name):
+                            return _mangle_private_name(obj, func, name)
```

Comment:
@victor-torres could you please check coverage report (https://codecov.io/gh/scrapy/scrapy/compare/36abe9235c7e85f89fe80ae726ca2a72b8ed7971...47a992615a046d20c31bfca6f8b65c3194e9fd30/diff) - it shows that this line is never executed. I'm not sure if that's coverage report which is wrong, but it worths investigating.

Reply:
Very interesting. It seems like this code is not needed anymore because we're getting the already mangled name when matching func with spider attributes.